libr/core/cmd_type.c

```diff
@@ -745,6 +746,40 @@ static int cmd_type(void *data, const char *input) {
 				r_core_cmd_help (core, help_msg_ts);
 			}
 			break;
+		case 'c': { //tsc
+			char *name = NULL;
+			char *varname = NULL;
+           		SdbKv *kv;
+           		SdbListIter *iter;
+			int i, j;
+        		SdbList *l = sdb_foreach_list (TDB, true);
+        		ls_foreach (l, iter, kv) {
+        			if (!strcmp (sdbkv_value (kv), "struct")) {
+                			if (!name || strcmp (sdbkv_value (kv), name)) {
+                				free (name);
+                    				name = strdup (sdbkv_key (kv));
+                    				const char *q = sdb_fmt ("struct.%s", name);
+                        			const char *res = sdb_const_get (TDB, q, 0);
+						if (res) {
+							r_cons_printf ("%s %s{", sdbkv_value (kv), name);
+							for (i = 0; i < sdb_array_length (TDB, q); i++) {
+								varname = sdb_array_get (TDB, q, i, 0);
+								const char *fulltypename = sdb_const_get (TDB, sdb_fmt ("%s.%s", q, varname), 0);
+								for (j = 0; (isalpha (fulltypename[j]) && (fulltypename[j] != ',')); j++) {
+									r_cons_printf ("%c", fulltypename[j]);
```

Comment:
use sdb_array_get instead of printing char by char until comma is found.

Reply:
char *typename = sdb_array_get(TDB, sdb_fmt ("%s.%s", q, varname),0, 0);
r_cons_printf ("%s %s;", typename, varname);
free (typename);

Reply:
like this